### PR TITLE
Language select

### PR DESCRIPTION
**Pull request by AWBaumann**
### Description
Changed the on-boarding language select to use radio buttons and factored the logic out so that it can be used elsewhere.

Then, I reused it elsewhere! You can now change the language from any applicant screen. Using the dropdown will refresh the current page in a different language.

image

### Checklist
- [x] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### Issue(s)
Fixes #<issue_number>; Fixes #<issue_number>; Fixes #<issue_number>...

### CODE REVIEW

File: universal-application-tool-0.0.1/app/views/LanguageUtils.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        langs.availables().stream().map(Lang::toLocale).collect(toImmutableList());
+  }
+
+  public Optional<String> getPreferredLangage(Http.Request request) {
+    Optional<Http.Cookie> language = request.cookies().get("PLAY_LANG");
+    if (language.isPresent()) {
+      return Optional.of(language.get().value());
+    }
+    return Optional.empty();
+  }
```
Comment (carolinedanzi) [inline universal-application-tool-0.0.1/app/views/LanguageUtils.java:41]: suggestion (blocking): Prefer the following to keep consistent with Play:

```java
// Inject MessagesApi in constructor, then:

Messages messages = messagesApi.preferred(request);
String locale = messages.lang().toString;
```

Need to finish  reading the rest, but using `Lang` or `Locale` seems safer than String - is there a strong reason to use String over a language type?

**Review by carolinedanzi — COMMENTED** [review]
This is super cool! Still reading through, but wanted to highlight one suggestion/question early

File: universal-application-tool-0.0.1/app/views/applicant/ApplicantLayout.java [hunk 2]
```diff
@@ -53,23 +64,40 @@ public Content render(HtmlBundle bundle) {
...
+    Tag csrfInput = input().isHidden().withValue(csrfToken).withName("csrfToken");
+
+    ContainerTag languageForm = div();
+    if (profile.isPresent()) { // Show language switcher.
+      long userId = Long.parseLong(profile.get().getId());
+      String updateLanguageAction =
+          controllers.applicant.routes.ApplicantInformationController.updateWithRedirect(
+                  userId, request.uri())
+              .url();
+      String preferredLanguage = languageUtils.getPreferredLangage(request).orElse("en");
```
Comment (carolinedanzi) [inline universal-application-tool-0.0.1/app/views/applicant/ApplicantLayout.java:89]: suggestion (blocking): Use our default language

```suggestion
      String preferredLanguage = languageUtils.getPreferredLangage(request).orElse(LocalizedStrings.DEFAULT_LOCALE.toLanguageTag());
```

**Review by carolinedanzi — COMMENTED** [review]
This looks really good! Are there any tests we could add you think would be helpful?

File: universal-application-tool-0.0.1/app/views/applicant/ApplicantLayout.java (hunk 2 above)
Comment (AWBaumann) [reply universal-application-tool-0.0.1/app/views/applicant/ApplicantLayout.java]: If I use `MessagesApi` like you suggested above then this kind of takes care of itself, doesn't it?

File: universal-application-tool-0.0.1/app/views/LanguageUtils.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+import javax.inject.Inject;
+import play.i18n.Lang;
+import play.i18n.Langs;
+import play.mvc.Http;
+import views.style.BaseStyles;
+import views.style.ReferenceClasses;
+import views.style.StyleUtils;
+import views.style.Styles;
+
+public class LanguageUtils {
```
[the remark this answers is not in the document]
Comment (AWBaumann) [reply universal-application-tool-0.0.1/app/views/LanguageUtils.java]: Done.

File: universal-application-tool-0.0.1/app/views/LanguageUtils.java (hunk 1 above)
Comment (AWBaumann) [reply universal-application-tool-0.0.1/app/views/LanguageUtils.java]: Everything downstream requires a String (selecting a specific value in a dropdown/radio list, changing the lang html attribute) but I think it's probably better to let those use cases handle it with `Lang.toString()`.

File: universal-application-tool-0.0.1/app/views/applicant/ApplicantLayout.java (hunk 2 above)
Comment (carolinedanzi) [reply universal-application-tool-0.0.1/app/views/applicant/ApplicantLayout.java]: Yes, this looks nice!

File: universal-application-tool-0.0.1/app/views/LanguageUtils.java (hunk 1 above)
Comment (carolinedanzi) [reply universal-application-tool-0.0.1/app/views/LanguageUtils.java]: Agreed, this looks nice!

**Review by carolinedanzi — APPROVED** [review]
Thank you Aaron!

**Pull request merged**